pandas/_libs/tslibs/offsets.pyx

```diff
@@ -1838,21 +1838,28 @@ cdef class YearOffset(SingleConstructorOffset):
     """
     DateOffset that just needs a month.
     """
-    _attributes = tuple(["n", "normalize", "month"])
 
-    # _default_month: int  # FIXME: python annotation here breaks things
+    _attributes = tuple(["n", "normalize", "month"])
+    _default_month = 1
 
-    cdef readonly:
-        int month
+    cdef:
+        readonly int month, _period_dtype_code
+        # public int _period_dtype_code
 
     def __init__(self, n=1, normalize=False, month=None):
         BaseOffset.__init__(self, n, normalize)
 
-        month = month if month is not None else self._default_month
-        self.month = month
+        if month is not None:
+            if isinstance(month, str):
+                self.month = MONTH_TO_CAL_NUM[month]
```

Comment:
this is adding support for passing a month string?  thats not necessarily a bad idea, but would need to check perf impact